Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+        if (userInfo != null && userInfo.getUserName() != null && userInfo.getUserName().length() > 0) {
+            userId = userInfo.getUserName();
+        }
+
+        if (userId != null) {
+            participant.setUserID(userId);
+        }
+
+        // If specified, set the User Name
+        String userName = null;
+        if (userInfo != null && userInfo.getPersonName() != null) {
+            if (userInfo.getPersonName().getGivenName() != null && userInfo.getPersonName().getGivenName().length() > 0) {
+                userName = userInfo.getPersonName().getGivenName();
+            }
+
+            if (userInfo.getPersonName().getFamilyName() != null
+                && userInfo.getPersonName().getFamilyName().length() > 0) {
+                if (userName != null) {
+                    userName += (" " + userInfo.getPersonName().getFamilyName());
+                } else {
+                    userName = userInfo.getPersonName().getFamilyName();
+                }
+            }
+        }
+        if (userName != null) {
+            participant.setUserName(userName);
+        }
+
+        // Set the UserIsRequester Flag
+        participant.setUserIsRequestor(userIsReq);
+        return participant;
+    }
+
+    public static EventIdentificationType getEventIdentificationType(CodedValueType eventID, boolean isRequesting) {
+        EventIdentificationType oEventIdentificationType = AuditDataTransformHelper.createEventIdentification(
+            isRequesting ? AuditDataTransformConstants.EVENT_ACTION_CODE_READ : AuditDataTransformConstants.EVENT_ACTION_CODE_CREATE,
+            AuditDataTransformConstants.EVENT_OUTCOME_INDICATOR_SUCCESS, eventID);
+        return oEventIdentificationType;
+    }
+
+    public static AuditMessageType.ActiveParticipant getActiveParticipantSource(boolean isRequesting) {
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_USER_ID_SOURCE);
+        participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        try {
+            String ipAddr = InetAddress.getLocalHost().getHostAddress();
+            participant.setNetworkAccessPointID(ipAddr);
+            participant.setNetworkAccessPointTypeCode(AuditDataTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_IP);
+        } catch (UnknownHostException ex) {
+            LOG.warn("UnknownHostException thrown getting local host address.", ex);
+            return null;
+        }
+
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_CDE, null,
+            AuditDataTransformConstants.EVENT_ID_CODE_SYS_NAME_DOC, CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(isRequesting ? Boolean.TRUE : Boolean.FALSE);
```

Comment:
unecessary if statement.  why are we checking the boolean instead of just setting: participant.setUserIsRequestor(isRequesting);